### PR TITLE
Adding snowflake table name option and regex

### PR DESCRIPTION
**Pull request by Luishfs**
# Summary
Added Snowflake Table Name option on io_config.yaml
Added `table_name_parts_from_query` option to `io.snowflake` if no Table name is specified
Updated docs

cc:
@wangxiaoyou1993

### CODE REVIEW

File: mage_ai/io/snowflake.py [hunk 1]
```diff
@@ -172,6 +176,10 @@ def load(
 
         query_string = self._clean_query(query_string)
 
+        if not table_name and not full_table_name:
+            # Try to find table name via query
+            table_name = table_name_parts_from_query(query_string)[2]
```
Comment (wangxiaoyou1993) [inline mage_ai/io/snowflake.py:181]: Please firstly check whether the return value of table_name_parts_from_query is None.

File: docs/design/data-loading.mdx [hunk 2]
```diff
@@ -978,8 +978,11 @@ SNOWFLAKE_DEFAULT_SCHEMA: null              # Optional default schema
 SNOWFLAKE_PRIVATE_KEY_PASSPHRASE: null      # Optional private key passphrase
 SNOWFLAKE_PRIVATE_KEY_PATH: null            # Optional private key path
 SNOWFLAKE_ROLE: null                        # Optional role name
+SNOWFLAKE_TABLE_NAME: null                  # Optional table name
```
Comment (wangxiaoyou1993) [inline docs/design/data-loading.mdx:981]: We don't need this config field

File: mage_ai/io/snowflake.py (hunk 1 above)
Comment (Luishfs) [reply mage_ai/io/snowflake.py]: Added check on c92a9a19635f3ec527768bebdfa3cc8a4d55613f

File: docs/design/data-loading.mdx (hunk 2 above)
Comment (Luishfs) [reply docs/design/data-loading.mdx]: Removed on c92a9a19635f3ec527768bebdfa3cc8a4d55613f

File: mage_ai/io/snowflake.py [hunk 3]
```diff
@@ -172,6 +176,12 @@ def load(
 
         query_string = self._clean_query(query_string)
 
+        if not table_name and not full_table_name:
+            # Try to find table name via query
+            table_name = table_name_parts_from_query(query_string)
```
Comment (wangxiaoyou1993) [inline mage_ai/io/snowflake.py:181]: nit: table_name_parts

**Review by wangxiaoyou1993 — APPROVED** [review]

**Pull request merged**